src/collections/componentSet.ts

```diff
@@ -269,11 +269,21 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
 
     const typeMap = new Map<string, string[]>();
 
-    const addToTypeMap = (typeName: string, fullName: string): void => {
-      if (!typeMap.has(typeName)) {
-        typeMap.set(typeName, []);
+    const addToTypeMap = (type: MetadataType, fullName: string): void => {
+      if (type.isAddressable !== false) {
+        const typeName = type.name;
+        if (!typeMap.has(typeName)) {
+          typeMap.set(typeName, []);
+        }
+        const typeEntry = typeMap.get(typeName);
+        if (fullName === ComponentSet.WILDCARD) {
+          typeMap.set(typeName, [fullName]);
+        } else {
+          if (!typeEntry.includes(fullName) && !typeEntry.includes(ComponentSet.WILDCARD)) {
+            typeMap.get(typeName).push(fullName);
+          }
```

Comment:
This conditional prevents a wildcard from being written in addition to fullNames.  It also prevents duplicates.